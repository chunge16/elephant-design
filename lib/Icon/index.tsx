import React from 'react';
import '../importIcons';

interface IconProps {
    name: string
}

const Icon: React.FunctionComponent<IconProps> = (props) => {
    const {name} = props;
    return (
        <span>
             <svg>
                 <use xlinkHref={`#${name}`}/>
             </svg>
        </span>
    );
};

export default Icon;
