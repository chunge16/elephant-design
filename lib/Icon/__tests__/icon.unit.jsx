import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../icon';
import {mount} from 'enzyme';

describe('icon', () => {
    it('render successfully', () => {
        const json = renderer.create(<Icon name="add"/>).toJSON();
        expect(json).toMatchSnapshot();
    });
    it('onClick', () => {
        const fn = jest.fn();
        const component = mount(<Icon name="wechat" onClick={fn}/>);
        component.find('svg').simulate('click');
        expect(fn).toBeCalled();
    });
});
