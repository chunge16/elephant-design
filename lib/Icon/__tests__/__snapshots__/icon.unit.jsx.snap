// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render successfully 1`] = `
<svg
  className=""
>
  <use
    xlinkHref="#add"
  />
</svg>
`;
